cranelift/frontend/src/frontend.rs

```diff
@@ -968,24 +969,21 @@ mod tests {
         sample_function(true)
     }
 
-    #[test]
-    #[cfg_attr(not(target_arch = "x86_64"), ignore)]
-    fn memcpy() {
-        use core::str::FromStr;
-        use cranelift_codegen::{isa, settings};
-
+    /// Helper function to construct the host's native target ISA.
+    fn host_target() -> Box<dyn TargetIsa> {
         let shared_builder = settings::builder();
         let shared_flags = settings::Flags::new(shared_builder);
+        let isa_builder = cranelift_native::builder().unwrap_or_else(|msg| {
+            panic!("host machine is not supported: {}", msg);
+        });
+        isa_builder.finish(shared_flags)
+    }
 
-        let triple =
-            ::target_lexicon::Triple::from_str("x86_64").expect("Couldn't create x86_64 triple");
-
-        let target = isa::lookup(triple)
-            .ok()
-            .map(|b| b.finish(shared_flags))
-            .expect("This test requires x86_64 support.");
-
-        let mut sig = Signature::new(target.default_call_conv());
+    #[cfg_attr(not(target_pointer_width = "64"), ignore)]
+    #[test]
+    fn memcpy() {
+        let target = host_target();
+        let mut sig = Signature::new(CallConv::SystemV);
```

Comment:
I think you could get the default calling convention from the `TargetIsa` object, and then use it to generate the calling convention in the expected string below. Not sure if it works for all the cases, though.

Reply:
Ok, I ended up changing the approach... I am just trying various things based on the discussion [here](https://bytecodealliance.zulipchat.com/#narrow/stream/217126-wasmtime/topic/testing).